index.js

```diff
@@ -19,100 +13,119 @@ const npar = spec => {
 cmd.restricted = (spec, opts) => setAccess(spec, 'restricted', opts)
-function setAccess (spec, access, opts) {
-  opts = AccessConfig(opts)
-  return pwrap(opts, () => {
-    spec = npar(spec)
-    validate('OSO', [spec, access, opts])
-    const uri = `/-/package/${eu(spec.name)}/access`
-    return npmFetch(uri, opts.concat({
-      method: 'POST',
-      body: { access },
-      spec
-    }))
-  }).then(res => res.body.resume() && true)
+function setAccess (spec, access, opts = {}) {
+  return new Promise((resolve, reject) => {
+    try {
+      spec = npar(spec)
+      validate('OSO', [spec, access, opts])
+      const uri = `/-/package/${eu(spec.name)}/access`
+      return npmFetch(uri, {
+        ...opts,
+        method: 'POST',
+        body: { access },
+        spec
+      })
+        .then(() => resolve(true))
+        .catch(reject)
+    } catch (err) {
+      return reject(err)
+    }
+  })
 }
 
-cmd.grant = (spec, entity, permissions, opts) => {
-  opts = AccessConfig(opts)
-  return pwrap(opts, () => {
-    spec = npar(spec)
-    const { scope, team } = splitEntity(entity)
-    validate('OSSSO', [spec, scope, team, permissions, opts])
-    if (permissions !== 'read-write' && permissions !== 'read-only') {
-      throw new Error('`permissions` must be `read-write` or `read-only`. Got `' + permissions + '` instead')
+cmd.grant = (spec, entity, permissions, opts = {}) => {
+  return new Promise((resolve, reject) => {
+    try {
+      spec = npar(spec)
+      const { scope, team } = splitEntity(entity)
+      validate('OSSSO', [spec, scope, team, permissions, opts])
+      if (permissions !== 'read-write' && permissions !== 'read-only') {
+        throw new Error('`permissions` must be `read-write` or `read-only`. Got `' + permissions + '` instead')
+      }
+      const uri = `/-/team/${eu(scope)}/${eu(team)}/package`
+      return npmFetch(uri, {
+        ...opts,
+        method: 'PUT',
+        body: { package: spec.name, permissions },
+        scope,
+        spec,
+        ignoreBody: true
+      })
+        .then(() => resolve(true))
+        .catch(reject)
+    } catch (err) {
+      return reject(err)
     }
-    const uri = `/-/team/${eu(scope)}/${eu(team)}/package`
-    return npmFetch(uri, opts.concat({
-      method: 'PUT',
-      body: { package: spec.name, permissions },
-      scope,
-      spec,
-      ignoreBody: true
-    }))
-  }).then(() => true)
+  })
 }
 
-cmd.revoke = (spec, entity, opts) => {
-  opts = AccessConfig(opts)
-  return pwrap(opts, () => {
-    spec = npar(spec)
-    const { scope, team } = splitEntity(entity)
-    validate('OSSO', [spec, scope, team, opts])
-    const uri = `/-/team/${eu(scope)}/${eu(team)}/package`
-    return npmFetch(uri, opts.concat({
-      method: 'DELETE',
-      body: { package: spec.name },
-      scope,
-      spec,
-      ignoreBody: true
-    }))
-  }).then(() => true)
+cmd.revoke = (spec, entity, opts = {}) => {
+  return new Promise((resolve, reject) => {
```

Comment:
I think this whole bit, with the try/reject and all, can be simplified to just replace `pwrap(...` with `Promise.resolve().then(...`

Then you can get rid of the then/catch on the `npmFetch` returned promise, and instead just return that promise.  Anything thrown in the function will reject the promise.